src/copy.rs

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2023-present, PingCAP, Inc. Licensed under Apache-2.0.
+
+use std::fs::{copy, create_dir_all};
+use std::os::unix::fs::symlink;
+use std::path::Path;
+use std::sync::Arc;
+
+use crate::config::Config;
+use crate::env::{FileSystem, OFlag};
+use crate::file_pipe_log::{FileNameExt, FilePipeLogBuilder};
+use crate::pipe_log::{FileId, LogQueue};
+
+/// Make a copy from `source` to `target`. `source` should exists but `target`
+/// shouldn't.
+///
+/// *minimum* means *symlink* will be used if possbile, otherwise *copy* will be
+/// used instead. Generally all inactive log files will be symlinked, but the
+/// last active one will be copied.
+///
+/// After the copy is made both of 2 engines can be started and run at the same
+/// time.
+pub fn minimum_copy<F, P>(cfg: &Config, fs: Arc<F>, target: P) -> Result<(), String>
+where
+    F: FileSystem,
+    P: AsRef<Path>,
+{
+    if cfg.enable_log_recycle {
```

Comment:
why? the error from recycle should be ignored.

Reply:
I thin minimun_copy is only allowed when `recycle` is disabled. Otherwise the target instance can't be run concurrently with the original one.

Reply:
But you can't disallow recycle when opening this db again. e.g. user change the config after creating a checkpoint.

Reply:
Users can also disable recycle in the config before creating a checkpoint. I think it lefts less chances to users to make mistakes.